packages/core/core/src/services/server/index.ts

```diff
@@ -15,9 +16,25 @@ const healthCheck: Core.MiddlewareHandler = async (ctx) => {
   ctx.status = 204;
 };
 
+const shouldTrustProxy = (strapi: Core.Strapi) => {
+  const explicit = strapi.config.get('server.proxy.koa', undefined);
+
+  if (isBoolean(explicit)) {
```

Comment:
We might want to just return when explicit===true, but when `explicit === false && implict` we output a warning that there is a conflict with the settings and that we're ignoring it because it would break the koa server.